low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +42,365 @@ class ServerMap(ResourceMap):
+            recommendation = 'None' if good_state\
+                             else "Fault detected, please contact Seagate support."
+
+        health_data["health"].update({
+            "status": status,
+            "description": description,
+            "recommendation": recommendation,
+            "specifics": specification
+        })
+
+    @staticmethod
+    def convert_cpu_info_list(cpu_info):
+        """
+        Converts cpu info as read from file to a list of cpu indexes
+
+        Example
+            '0-2,4,6-8' -> [0,1,2,4,6,7,8]
+        """
+        # Split the string with comma
+        cpu_info = cpu_info.split(',')
+        cpu_list = []
+        for item in cpu_info:
+            # Split item with a hyphen if it is a range of indexes
+            item = item.split('-')
+            if len(item) == 2:
+                # Item is a range
+                num1 = int(item[0])
+                num2 = int(item[1])
+                # Append all indexes in that range
+                for i in range(num1, num2+1):
+                    cpu_list.append(i)
+            elif len(item) == 1:
+                # Item is a single index
+                cpu_list.append(int(item[0]))
+        return cpu_list
+
+    def get_cpu_list(self, mode):
+        """Returns the cpus present after reading."""
+        cpu_info_path = Path(self.cpu_path+mode)
+        # Read the text from /cpu/online file
+        cpu_info = cpu_info_path.read_text()
+        # Drop the \n character from the end of string
+        cpu_info = cpu_info.rstrip('\n')
+        # Convert the string to list of indexes
+        cpu_list = self.convert_cpu_info_list(cpu_info)
+        return cpu_list
+
+    def get_cpu_info(self):
+        """Update and return CPU information in specific format."""
+        cpu_data = []
+        cpu_present = self.get_cpu_list("present")
+        cpu_online = self.get_cpu_list("online")
+        cpu_usage = psutil.cpu_percent(interval=None, percpu=True)
+        cpu_usage_dict = dict(zip(cpu_online, cpu_usage))
+
+        for cpu_id in range(0, len(cpu_present)):
+            uid = f"cpu_{cpu_id}"
+            cpu_dict = self.get_data_template(uid, False)
+            online_status = "Online" if cpu_id in cpu_online else "Offline"
+            health_status = "OK" if online_status == "Online" else "Fault"
+            usage = "NA" if health_status == "Fault" \
+                else cpu_usage_dict[cpu_id]
+            specification = [
+                {
+                    "cpu_usage": usage,
+                    "state": online_status
+                }
+            ]
+            self.set_health_data(cpu_dict, status=health_status,
+                                 specification=specification)
+            cpu_data.append(cpu_dict)
+        return cpu_data
+
+    def get_disks_info(self):
+        """Update and return server drive information in specific format."""
+        return self._get_drives()
+
+    def _get_drives(self):
```

Comment:
seems code is duplicated from disk sensor. Better would be to add these common functions in common utilities and use from there, so later sensor also we can  update to use from there